From b17d194bd8285a19382a902a0bec9e5e042df064 Mon Sep 17 00:00:00 2001
From: Nobuhiro Iwamatsu <nobuhiro1.iwamatsu@toshiba.co.jp>
Date: Tue, 16 Apr 2019 08:52:01 +0900
Subject: [PATCH 2/4] Add support GNUInstallDirs

This adds the functionality of the module "GNUInstallDirs" to make the
installation compatible with GNU.

https://cmake.org/cmake/help/v3.14/module/GNUInstallDirs.html

Signed-off-by: Nobuhiro Iwamatsu <nobuhiro1.iwamatsu@toshiba.co.jp>
---
 CMakeLists.txt     | 2 ++
 src/CMakeLists.txt | 8 ++++----
 2 files changed, 6 insertions(+), 4 deletions(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index 104969e..57477fc 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -10,6 +10,8 @@ add_definitions(-DVERSION="${VERSION}")
 
 set(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} -std=gnu99")
 
+include("GNUInstallDirs")
+
 #set(CMAKE_C_FLAGS_DEBUG "-g")
 include_directories ("${PROJECT_SOURCE_DIR}/src")
 add_subdirectory (src)
diff --git a/src/CMakeLists.txt b/src/CMakeLists.txt
index ea5979c..d97f221 100644
--- a/src/CMakeLists.txt
+++ b/src/CMakeLists.txt
@@ -19,7 +19,7 @@ add_executable(fw_setenv fw_setenv.c)
 target_link_libraries(fw_printenv ubootenv z)
 target_link_libraries(fw_setenv ubootenv z)
 
-install (TARGETS ubootenv DESTINATION lib)
-install (FILES libuboot.h DESTINATION include)
-install (TARGETS fw_printenv DESTINATION bin)
-install (TARGETS fw_setenv DESTINATION bin)
+install (TARGETS ubootenv DESTINATION "${CMAKE_INSTALL_LIBDIR}")
+install (FILES libuboot.h DESTINATION "${CMAKE_INSTALL_INCLUDEDIR}")
+install (TARGETS fw_printenv DESTINATION "${CMAKE_INSTALL_BINDIR}")
+install (TARGETS fw_setenv DESTINATION "${CMAKE_INSTALL_BINDIR}")
-- 
2.20.1

